lib/datadog/tracing/contrib/stripe/request.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+require_relative '../../metadata/ext'
+require_relative '../../analytics'
+require_relative 'ext'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module Stripe
+        # Defines instrumentation for Stripe requests
+        module Request
+          module_function
+
+          def start_span(event)
+            # Start a trace
+            Tracing.trace(Ext::SPAN_REQUEST).tap do |span|
+              event.user_data[:datadog_span] = span
+            end
+          end
+
+          def finish_span(event)
+            event.user_data[:datadog_span].tap do |span|
+              # If no active span, return.
+              return nil if span.nil?
+
+              tag_span(span, event)
+
+              # Finish the span
+              span.finish
+            end
+          end
+
+          def tag_span(span, event)
+            span.resource = event.object_name if event.respond_to?(:object_name) && event.object_name
```

Comment:
This particular resource configuration is dependent upon https://github.com/stripe/stripe-ruby/pull/1168, which exposes `object_name`. I'm happy to remove this while that PR is considered if it's preferred!

Reply:
I think it would be great if we could derive `resource` from Stripe's api resource (with your Stripe's PR). 

However, I feel like having `stripe.customer` instead of `customer` would be an improvement (currently, without an object name, it is `stripe.request`). What do you think?

Reply:
Yep that makes a lot of sense, @TonyCTHsu. I've updated this to be `"stripe.#{event.object_name}"`.

I have not yet heard back from Stripe on that PR, so I'm curious what you suggest doing while we wait. I'm happy to remove this to unblock the rest of this instrumentation. Then I can open a new PR here once stripe/stripe-ruby#1168 (hopefully) gets merged.

Reply:
I think we can merge this with a code comment referencing the Stripe's PR. What bothers me more would be a missing spec to test this condition.

Reply:
@TonyCTHsu makes sense! I've just pushed up a test that I think best exercises this. I can keep an eye on stripe/stripe-ruby#1168, and update as appropriate based on what happens with that.